src/main/java/co/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -101,7 +103,19 @@ public int runBasicQueries(RestClient client, DiagnosticContext context) throws
     */
     public void getAllPages(RestClient client, List<RestEntry> queries, int perPage, RestEntry action) throws DiagnosticException {
         // get the values needed to the pagination.
-        RestResult res = client.execQuery(String.format("%s?per_page=1", action.getUrl()));
+        RestResult res;
+        Pattern p = Pattern.compile( ".*\\?.*" );
+        Matcher m = p.matcher( action.url );
+        if (m.find()) {
+            res = client.execQuery(String.format("%s&per_page=1", action.getUrl()));
+        }
+        else {
+            if (action.getName().equals("kibana_fleet_agents") || action.getName().equals("kibana_fleet_agent_policies") || action.getName().equals("kibana_fleet_package_policies")) {
+                res = client.execQuery(String.format("%s?perPage=1", action.getUrl()));
+            } else {
+                res = client.execQuery(String.format("%s?per_page=1", action.getUrl()));
+            }
+        }
```

Comment:
This code is modifying pre-existing bad code, but instead of running a regex against the URL to find if it has a `?` in it, let's explicitly parse it in a function so we can make better sense of it, and not make as many assumptions about future versions of the URL (if they fix the `perPage` difference, we'll want to rename the action for fixed versions instead of updating code):

```java
private void getPageUrl(RestEntry action, int page, int perPage) {
  String actionUrl = action.getUrl();
  String perPageField = "per_page";

  if (
    action.getName().equals("kibana_fleet_agents") ||
    action.getName().equals("kibana_fleet_agent_policies") ||
    action.getName().equals("kibana_fleet_package_policies")
  ) {
    perPageField = "perPage";
  }

  final String querystringPrefix = URI.create(actionUrl).getQuery() != null ? "?" : "&";
  final String params = "page=" + page + "&" + perPageField + "=" + perPage;

  return actionUrl + querystringPrefix + params;
}
```

Then this code could be replaced with:

```java
RestResult res = client.execQuery(getPageUrl(action, 1, 1));
```

Ideally we'd actually use the response for more than just looking up the totals, then asking for the first page _again_, but that's largely what I mean about pre-existing bad code.

Reply:
FYI, I created an issue around the idea of different `perPage` parameters for Kibana. https://github.com/elastic/kibana/issues/135490